### PR TITLE
Added OpenStack support for safety controller

### PR DESCRIPTION
**Pull request by prashanth26**
This PR fills up the interfaces required to support safety controller for openstack to avoid creating orphan VMs.

### CODE REVIEW

File: pkg/controller/machine_safety.go [hunk 1]
```diff
@@ -141,8 +141,9 @@ func (c *controller) checkAndFreezeORUnfreezeMachineSets(wg *sync.WaitGroup) {
 func (c *controller) checkVMObjects(wg *sync.WaitGroup) {
 	var wgForCheckVM sync.WaitGroup
 
-	wgForCheckVM.Add(1)
+	wgForCheckVM.Add(2)
 	go c.checkAWSMachineClass(&wgForCheckVM)
+	go c.checkOSMachineClass(&wgForCheckVM)
```
Comment (hardikdr) [inline pkg/controller/machine_safety.go]: Would be nice to have this as part of the doku on how to add support for new provider.

File: pkg/controller/machine_safety.go [hunk 2]
```diff
@@ -358,7 +382,8 @@ func (c *controller) deleteOrphanVM(vm driver.VM, secretRef *corev1.Secret, kind
 	err := dvr.Delete()
 	if err != nil {
 		glog.Warning("Error while deleting VM on CP - ", err)
```
Comment (hardikdr) [inline pkg/controller/machine_safety.go]: Can we have it as Error and not Warning. Also please describe in the logs that, controller will retry after resync to delete it again.

File: pkg/apis/machine/validation/openstackmachineclass.go [hunk 3]
```diff
@@ -58,6 +60,30 @@ func validateOpenStackMachineClassSpec(spec *machine.OpenStackMachineClassSpec,
...
+	return allErrs
+}
+
+func validateOSClassSpecTags(tags map[string]string, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	clusterName := ""
+	nodeRole := ""
+
+	for key := range tags {
+		if strings.Contains(key, "kubernetes.io-cluster-") {
```
[the remark this answers is not in the document]
Comment (prashanth26) [reply pkg/apis/machine/validation/openstackmachineclass.go]: Done.

File: pkg/controller/machine_safety.go (hunk 1 above)
Comment (prashanth26) [reply pkg/controller/machine_safety.go]: Done.

**Comment by prashanth26** [comment]
I have made requested changes. Please check the latest commit for changes.

**Pull request merged**